pkg/controller/replicaset/metrics/metrics.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"sync"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+const ReplicaSetControllerSubsystem = "replicaset_controller"
+
+var DeletedPodAgeRatio = metrics.NewHistogram(
+	&metrics.HistogramOpts{
```

Comment:
Do you really want to use the default buckets for your use case? They are meant for web services latency, so I don't think they should be a good fit for your use case.

Reply:
Good point, should I just manually define these buckets then as something like `[0.0, 2.0]`? If we are trying to measure the amount under `2`, does that make sense? I'm just not clear on the code syntax for defining these buckets, and couldn't find any good docs on it

Reply:
If the sole purpose of this metric is to measure all the operations done in less than 2s, then I don't think a histogram is the best fit. The best option for this case would be to use a counter accounting for all the operations that were completed in under 2 seconds. However, if you need more granularity on the duration of the operation, then a histogram will be a good fit.

As for defining custom buckets, you will need to think about the different time ranges that are meaningful to be collected for your use case.  For instance, the apiserver admission controller/webhook latency metrics are using these [custom buckets](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/admission/metrics/metrics.go#L48-L49).

Reply:
The metric is about checking how far from 2s we get. Then, the range should be something like `[0, 8]`.

Reply:
So do we want several buckets on either side of 2? Like `[0, 0.5, 1, 1.5, 2, 2.5, 3, 4, 6, 8]`?

Reply:
Can we use exponential buckets? I would start at 0.25 with a factor of 2.

Reply:
Yes, from an instrumentation perspective, I think it would make sense to have [exponential buckets](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/component-base/metrics/histogram.go#L33-L36) here. Starting from 0.25 with a factor of 2, until you reach 8, so you will have a count of 6 and end up with the following buckets `[0.25, 0.5, 1, 2, 4, 8]`.

Reply:
Ok, thanks for the help everyone. I have pushed with the new buckets